exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -15,88 +15,152 @@
 package splunkhecexporter
 
 import (
-	"errors"
 	"fmt"
 	"math"
-	"strconv"
 
-	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/translator/conventions"
-	"go.opentelemetry.io/collector/translator/internaldata"
 	"go.uber.org/zap"
-	"google.golang.org/protobuf/types/known/timestamppb"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/splunk"
 )
 
 const (
 	// unknownHostName is the default host name when no hostname label is passed.
 	unknownHostName = "unknown"
-	// separator for metric values.
-	separator = "."
 	// splunkMetricValue is the splunk metric value prefix.
 	splunkMetricValue = "metric_name"
-	// bucketSuffix is the bucket suffix for distribution buckets.
-	bucketSuffix = "bucket"
-	// quantileSuffix is the quantile suffix for summary quantiles.
-	quantileSuffix = "quantile"
 	// countSuffix is the count metric value suffix.
 	countSuffix = "count"
-	// sumOfSquaredDeviation is the sum of squared deviation metric value suffix.
-	sumOfSquaredDeviation = "sum_of_squared_deviation"
-)
-
-var (
-	// infinity bound dimension value is used on all histograms.
-	infinityBoundSFxDimValue = float64ToDimValue(math.Inf(1))
 )
 
 func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config) ([]*splunk.Event, int, error) {
```

Comment:
Is it possible to split this func into smaller funcs to make it more readable? The nesting of fors, switches and ifs is hard to parse.